diff -Nru mame_base/src/emu/sound/scsp.c mame/src/emu/sound/scsp.c
--- mame_base/src/emu/sound/scsp.c	2007-11-18 15:54:28.000000000 -0800
+++ mame/src/emu/sound/scsp.c	2007-11-30 19:44:49.000000000 -0800
@@ -38,7 +38,7 @@
 #define FIX(v)	((UINT32) ((float) (1<<SHIFT)*(v)))
 
 
-#define EG_SHIFT	8
+#define EG_SHIFT	16
 
 // include the LFO handling code
 #include "scsplfo.c"
@@ -108,6 +108,7 @@
 					920.0,790.0,690.0,550.0,460.0,390.0,340.0,270.0,230.0,200.0,170.0,140.0,110.0,98.0,85.0,68.0,57.0,49.0,43.0,34.0,
 					28.0,25.0,22.0,18.0,14.0,12.0,11.0,8.5,7.1,6.1,5.4,4.3,3.6,3.1};
 static UINT32 FNS_Table[0x400];
+static INT32 EG_TABLE[0x400];
 
 typedef enum {ATTACK,DECAY1,DECAY2,RELEASE} _STATE;
 struct _EG
@@ -378,7 +379,7 @@
 	else
 		rate=0; //rate=((FNS(slot)>>9)&1);
 
-	slot->EG.volume=0;
+	slot->EG.volume=0x17F<<EG_SHIFT;
 	slot->EG.AR=Get_AR(SCSP,rate,AR(slot));
 	slot->EG.D1R=Get_DR(SCSP,rate,D1R(slot));
 	slot->EG.D2R=Get_DR(SCSP,rate,D2R(slot));
@@ -410,7 +411,7 @@
 			break;
 		case DECAY1:
 			slot->EG.volume-=slot->EG.D1R;
-			if(slot->EG.volume>>(EG_SHIFT+5)<=slot->EG.DL)
+			if(slot->EG.volume>>(EG_SHIFT+5)<slot->EG.DL)
 				slot->EG.state=DECAY2;
 			break;
 		case DECAY2:
@@ -426,7 +427,7 @@
 			if(slot->EG.volume<=0)
 			{
 				SCSP_StopSlot(slot,0);
-				slot->EG.volume=0;
+				slot->EG.volume=0x17F<<EG_SHIFT;
 				slot->EG.state=ATTACK;
 			}
 			break;
@@ -467,7 +468,7 @@
 	slot->step=SCSP_Step(slot);
 	Compute_EG(SCSP,slot);
 	slot->EG.state=ATTACK;
-	slot->EG.volume=0;
+	slot->EG.volume=0x17F<<EG_SHIFT;
 	slot->Prev=0;
 	slot->Backwards=0;
 	Compute_LFO(slot);
@@ -535,6 +536,13 @@
 		float fcent=44100.0*(1024.0+(double) i)/1024.0;
 		FNS_Table[i]=(float) (1<<SHIFT) *fcent;
 	}
+	
+	for(i=0;i<0x400;++i)
+	{
+		float envDB=((float)(3*(i-0x3ff)))/32.0;
+		float scale=(float)(1<<SHIFT);
+		EG_TABLE[i]=(INT32)(pow(10.0,envDB/20.0)*scale);
+	}
 
 	for(i=0;i<0x10000;++i)
 	{
@@ -1257,7 +1265,7 @@
 	if(!STWINH(slot))
 		*RBUFDST=sample;
 
-	sample=(sample*EG_Update(slot))>>SHIFT;
+	sample=(sample*EG_TABLE[EG_Update(slot)>>(SHIFT-10)])>>SHIFT;
 
 	return sample;
 }
@@ -1308,8 +1316,8 @@
 			if(EFSDL(slot))
 			{
 				unsigned short Enc=(TL(slot))|((EFPAN(slot))<<0x8)|((EFSDL(slot))<<0xd);
-				smpl+=(SCSP->DSP.EFREG[i]*SCSP->LPANTABLE[Enc])>>(SHIFT-5);
-				smpr+=(SCSP->DSP.EFREG[i]*SCSP->RPANTABLE[Enc])>>(SHIFT-5);
+				smpl+=(SCSP->DSP.EFREG[i]*SCSP->LPANTABLE[Enc])>>(SHIFT-4);
+				smpr+=(SCSP->DSP.EFREG[i]*SCSP->RPANTABLE[Enc])>>(SHIFT-4);
 			}
 		}
 
